virtualmachine/aws/objectstore_test.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"libretto/virtualmachine/common"
+)
+
+// Checks for an error in fetching bucket list and the functions return type
+// to be a slice of strings
+
+func TestGetS3BucketsList(t *testing.T) {
+	s3 := S3{
+		Name:   "",
+		Region: "us-east-1",
+		Prefix: "",
+	}
+
+	list, err := s3.GetS3BucketsList()
+
+	if err != nil {
+		t.Fail()
+	} else if reflect.TypeOf(list).Elem().Kind() != reflect.String {
+		t.Fail()
+	}
+
+	fmt.Println(list)
+}
+
+func TestCreateBucket(t *testing.T) {
+	bucketName := common.RandStringRunes(5)
+
+	s3 := S3{
+		Name:   bucketName,
+		Region: "us-east-1",
+		Prefix: "",
+	}
+
+	fmt.Println("Creating bucket with name %s", bucketName)
+	err := s3.CreateBucket()
+	if err != nil {
+		fmt.Println(err)
+		t.Fail()
+	}
+
+	list, err := s3.GetS3BucketsList()
+	found := false
+
+	for _, val := range list {
+		if val == bucketName {
+			found = true
+		}
+	}
+
+	if !found {
+		fmt.Println("Bucket not found")
+		t.Fail()
+	}
+
+	fmt.Println("Deleting bucket with name %s", bucketName)
+	err = s3.DeleteBucket()
+	if err != nil {
+		fmt.Println("Failed in deleting bucket")
+		t.Fail()
+	}
+
```

Comment:
we should check the list is not listing the bucket any more -- and we can make following a common function:

+	list, err := s3.GetS3BucketsList()
￼ +	found := false
￼ +
￼ +	for _, val := range list {
￼ +		if val == bucketName {
￼ +			found = true
￼ +		}
￼ +	}
return found